### DISCUSSION TITLE
Add Existing Virtual Network to reference with REUSE = True

### DISCUSSION THREAD

**Discussion by danielkelly22** [aztfmod/terraform-azurerm-caf#1409] [Ideas]
Is there any way to add in an existing VNet so that it can be referenced through the code, just like we can use and reference existing resource_groups like shown below?

`
resource_groups = {	
    m2_region1 = {
      name = "existing-rg"
	  reuse = true	
    }
  
  m2_networking_region1 = {
	name = "existing-network-rg"
	reuse = true		
  }	
}	`

It makes things very difficult to deploy resources using CAF to an existing virtual network.  It also adds steps and complexity when you want to deploy resources to subnets created with CAF because you have to enter in the subnet_id, but that won't exist until CAF runs so you have to build out in stages to be able to build resources (like load balancers,etc).  It would be very helpful to be able to pull in an existing vnet and be able to reference vnet_key and subnet_key to deploy everything at once.

**Comment by brk3** [comment]
The short answer is, no.

Referencing existing resources is something that is commonly asked for, but to date I'm unsure what the CAF maintainers view is on this. 

You could copy the `resource_group_reused` module pattern to do the same for vnets. I can foresee pitfalls with this but it's worth exploring to see if it helps your use case.

There is also Azure/caf-terraform-landingzones/pull/349 and aztfmod/terraform-azurerm-caf/pull/1068 which aim to solve this for all resource types which would be worth trying out also.
